src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp

```diff
@@ -460,8 +519,11 @@ bool EDP::validMatching(
     {
         return false;
     }
-    if (wdata->typeName() != rdata->typeName())
+
+    // Type Consistency Enforcement QosPolicy
+    if (!checkTypeValidation(wdata, rdata))
     {
+        // TODO Trigger INCONSISTENT_TOPIC status change
```

Comment:
Is it necessary  this TODO?

Reply:
Yes, as the standard stablishes:

> If either Step 1 or Step 2 fails, then the Topics associated with the DataWriter and DataReader are considered to be inconsistent: the DataWriter and DataReader shall not  communicate with each other, and the Service shall trigger an INCONSISTENT_TOPIC status change for both the DataReader’s Topic and the DataWriter’s Topic .

But at this point of the implementation we have no such as thing of status change, didn't we?